user_tasks/signals.py

```diff
@@ -0,0 +1,174 @@
+# -*- coding: utf-8 -*-
+"""Celery signal handlers."""
+
+from __future__ import absolute_import, unicode_literals
+
+import logging
+from uuid import uuid4
+
+from django.contrib.auth import get_user_model
+from django.utils.module_loading import import_string
+
+from celery.signals import before_task_publish, task_failure, task_prerun, task_retry, task_success
+
+from .exceptions import TaskCanceledException
+from .models import UserTaskStatus
+from .tasks import UserTaskMixin
+
+LOGGER = logging.getLogger(__name__)
+
+
+@before_task_publish.connect
+def create_user_task(sender=None, body=None, **kwargs):  # pylint: disable=unused-argument
+    """
+    Create a :py:class:`UserTaskStatus` record for each :py:class:`UserTaskMixin`.
+
+    Also creates a :py:class:`UserTaskStatus` for each chain, chord, or group containing
+    the new :py:class:`UserTaskMixin`.
+    """
+    try:
+        task_class = import_string(sender)
+    except ImportError:
+        return
```

Comment:
Thought I'd added a code comment regarding that, but looks like it got lost along the line.  There's a `celery.chord_unlock` task emitted during chord processing which doesn't actually resolve.  I suppose it's better to just handle that specific case rather than giving all ImportErrors a free pass, I was just erring on the side of not breaking things in a working Celery installation due to installing this app.